app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -168,9 +168,9 @@ def update_invalid_user
     current_user.increment_second_factor_attempts_count!
   end
 
-  def handle_valid_verification_for_confirmation_context
-    user_session[:authn_at] = Time.zone.now
-    @next_mfa_setup_path = next_setup_path
+  def handle_valid_verification_for_confirmation_context(auth_method:)
+    user_session[:auth_method] = auth_method
+    mark_user_session_authenticated(:valid_2fa_confirmation)
```

Comment:
Also, I assume this may cause some additional analytics to be logged via `mark_user_session_authenticated`'s call to `mark_user_session_authenticated_analytics`. Should we have specs for that?

Reply:
Yep, they were causing a failure, added a few changes to add specs around it.